### PR TITLE
Improve Yandex support

### PR DESCRIPTION
**Pull request by stevebissett**
I have fully documented the issue here: #397 (_Inconsistent behaviour when using Yandex on different devices)_ 

This serves as an example of what I view as the expected behaviour. 
On the parts that could be ambiguous, I have left some GitHub comments. 

Please feel free to discuss, challenge or engage on any of the ideas here :) 
I would like to ask that you read the whole issue in #397 before reviewing though.

### CODE REVIEW

File: test/unit/chrome_test.rb [hunk 1]
```diff
@@ -45,22 +45,6 @@ class ChromeTest < Minitest::Test
...
   end
 
-  test "detects yandex browser" do
-    browser = Browser.new(Browser["YANDEX_BROWSER"])
-
-    assert browser.yandex?
-    assert browser.chrome?
-    refute browser.safari?
-    assert browser.webkit?
-    assert_equal "41.0.2272.118", browser.full_version
```
Comment (stevebissett) [inline test/unit/chrome_test.rb]: This takes the underlying Chrome version, as opposed to the Yandex version (which makes sense if we are considering the browser to be chrome).

File: test/unit/yandex_browser_test.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+require "test_helper"
+
+class YandexTest < Minitest::Test
+  test "detects Yandex on iOS device" do
+    browser = Browser.new(Browser["YANDEX_BROWSER_IOS"])
+    assert browser.yandex?
+    assert browser.yandex_browser?
+    refute browser.safari?
+    refute browser.chrome?
```
Comment (stevebissett) [inline test/unit/yandex_browser_test.rb]: I am not strongly opinionated here on whether this should be true/false, however I lean more towards the side of false. Why? 

I would say the following statements are true: 

Yandex is a web browser
Yandex is my browser shown by https://www.whatismybrowser.com (when using Yandex, that is)
Yandex is not the same browser as Chrome
Yandex is based on Chromium

If I read the methods, I would assume that: 

`chrome?` is equivalent to saying `is_chrome` (standard in some other programming languages) 
`chromium_based?` is equivalent to saying `is_chromium_based` (is based on OSS Chromium) 
`webkit_based?` is equivalent to saying `is_webkit_based` (is based on the WebKit browser engine)

Comment (fnando) [reply test/unit/yandex_browser_test.rb]: I think it's more than that. What if I want to display a Chrome extension installer (either directly or via Google Play store) for Yandex users? If you drop the `chrome?` predicate then you have to do this for all browsers that are heavily based on Chrome. I'd bring back this behavior and if you need to do something specific for Yandex, go `yandex?` instead.

Does that make sense?

Comment (fnando) [reply test/unit/yandex_browser_test.rb]: Now, thinking a little bit about this, I don't know if I'm following this rule for every Chrome-based browser we're detecting. That said, it's probably fine `chrome?` to return `false`. At some point, make we should have something like `chrome_based?`.

**Pull request closed without merging**